### PR TITLE
[ADD] Add field to test is account cancel is installed to prevent to …

### PR DESCRIPTION
**Pull request by vrenaville**
Add field to test is account cancel is installed to prevent to have 2 cancel button on statement
It's the PR refer to comment on #259

### CODE REVIEW

File: account_constraints/model/account_bank_statement.py [hunk 1]
```diff
@@ -62,3 +62,14 @@ def process_reconciliation(self, mv_line_dicts):
         self = self.with_context(from_parent_object=True)
         return super(AccountBankStatementLine, self)\
             .process_reconciliation(mv_line_dicts)
+
+    @api.multi
+    def _is_account_cancel_installed(self):
+        ir_module = self.env['ir.module.module']
+        account_cancel = ir_module.search([('name', '=', 'account_cancel'),
+                                           ('state', '=', 'installed')])
+        return bool(account_cancel)
```
Comment (gurneyalex) [inline account_constraints/model/account_bank_statement.py]: you could use: 

```
return ir_module.search_count([('name', '=', 'account_cancel'),
               ('state', '=', 'installed')])
```

(replaces 69->71)

**Comment by gurneyalex** [comment]
:+1:

**Comment by sbidoul** [comment]
:+1: (code review, no test)

File: account_constraints/view/account_bank_statement.xml [hunk 2]
```diff
@@ -13,7 +13,8 @@
             <field name="arch" type="xml">
                 <xpath expr="//field[@name='bank_account_id']" position="after">
                     <field name="state" invisible="1"/>
-                    <button name="cancel" attrs="{'invisible': ['|', ('journal_entry_id', '=', False), ('state', '=', 'confirm')]}" string="Cancel" type="object" icon="gtk-undo"/>
+                    <field name='account_cancel_installed' invisible="1"/>
+                    <button name="cancel" attrs="{'invisible': ['|', ('journal_entry_id', '=', False),('account_cancel_installed','=',True) ,('state', '=', 'confirm')]}" string="Cancel" type="object" icon="gtk-undo"/>
```
Comment (guewen) [inline account_constraints/view/account_bank_statement.xml]: I think the domain should be:

```
['|', '|', ('journal_entry_id', '=', False), ('account_cancel_installed', '=', True) ,('state', '=', 'confirm')]
```

Your domain has a `AND` between the 2 last conditions.

**Comment by adrienpeiffer** [comment]
:+1: (Code review)

**Comment by guewen** [comment]
:-1: see my comment

**Comment by vrenaville** [comment]
Hello,

Thanks for the Review, I have update to code to fit the same domain as it is defined in account_cancel

**Comment by mdietrichc2c** [comment]
:+1:

**Comment by guewen** [comment]
Thanks for the change @vrenaville. The `('state', '=', 'confirm')` part in the domain seems useless in v8. So :+1: finally.

**Pull request merged**